shared/actions/settings.tsx

```diff
@@ -433,6 +434,27 @@ const loadLockdownMode = state =>
     )
     .catch(() => SettingsGen.createLoadedLockdownMode({status: null}))
 
+const loadProxyData = state =>
+  RPCTypes.configGetProxyDataRpcPromise(undefined)
+    .then((result: RPCTypes.ProxyData) => SettingsGen.createLoadedProxyData({proxyData: result}))
+    .catch(err => logger.warn('Error in loading proxy data', err))
+
+const saveProxyData = (_, proxyDataPayload: SettingsGen.SaveProxyDataPayload) =>
+  RPCTypes.configSetProxyDataRpcPromise(proxyDataPayload.payload).catch(err =>
+    logger.warn('Error in saving proxy data', err)
+  )
+
+const setCertificatePinning = (_, certPinningPayload: SettingsGen.SetCertificatePinningPayload) =>
+  RPCTypes.configGetProxyDataRpcPromise(undefined)
+      .then((result: RPCTypes.ProxyData) => {
+        const cloned = JSON.parse(JSON.stringify(result))
```

Comment:
you'll lose all typing by doing this, instead
```
const proxyData = {
...result,
certPinning: action.payload.enabled // also rename it to action so it matches everything else
}
```

Reply:
Deleted (sorry about that, that code was dead and I forgot to delete it).